src/main.rs

```diff
@@ -0,0 +1,77 @@
+mod dockerfile;
+
+use dockerfile::{Dockerfile, ValidationResult};
+use glob::glob;
+use std::{error::Error, process};
+
+fn main() {
+    let dockerfiles = get_dockerfiles().expect("[ERROR] Failed to get Dockerfiles.");
+    if dockerfiles.is_empty() {
+        skip();
+    };
+
+    let invalid = validate_dockerfiles(&dockerfiles);
+    if invalid.is_empty() {
+        pass();
+    }
+
+    fail(&invalid);
+}
+
+fn get_dockerfiles() -> Result<Vec<Dockerfile>, Box<dyn Error>> {
+    glob("**/*Dockerfile*")?
+        .map(|path| Dockerfile::new(path?))
+        .collect()
+}
+
+fn validate_dockerfiles(dockerfiles: &[Dockerfile]) -> Vec<(&Dockerfile, ValidationResult)> {
+    dockerfiles
+        .iter()
+        .map(|dockerfile| (dockerfile, dockerfile.validate()))
+        .filter(|result| !result.1.invalid_images.is_empty())
+        .collect()
+}
+
+fn skip() {
+    println!("[PASS] No Dockerfiles found.");
+    process::exit(0);
+}
+
+fn pass() {
+    println!("[PASS] All Dockerfiles are using versioned images.");
+    process::exit(0);
+}
+
+fn fail(results: &[(&Dockerfile, ValidationResult)]) {
+    println!("[FAIL] Found Dockerfiles not using versioned images.");
+    results.iter().for_each(|result| {
```

Comment:
Could also destructure here:
```suggestion
    results.iter().for_each(|(dockerfile, result)| {
```